curvefs/src/client/s3/client_s3_cache_manager.cpp

```diff
@@ -1309,8 +1430,11 @@ void DataCache::Write(uint64_t chunkPos, uint64_t len, const char *data,
             memcpy(newDatabuf, data, len);
             memcpy(newDatabuf + len, data_ + (chunkPos + len - chunkPos_),
                    totalSize - len);
+            chunkCacheManager_->rwLockWrite_.WRLock();
             Swap(newDatabuf, totalSize);
             chunkPos_ = chunkPos;
+            chunkCacheManager_->UpdateWrteCacheMap(oldChunkPos);
+            chunkCacheManager_->rwLockWrite_.Unlock();
```

Comment:
why not remove all overlap data cache including the current one from chunkcachemanager firstly, then put it back to the manager after writing

Reply:
I didn't understand

Reply:
if I understand correctly, `ChunkCacheManager::FindWriteableDataCache` returns the first datacache in ChunkCacheManager that overlaps with current write operation and removes all following overlap datachche from ChunkCacheManager.

so, why not remove the first overlap data cache together and put it back after finishing the write operation. In this way, data cache doesn't have to care about start offset is changed, and  hold ChunkCacheManager's lock to handle this case.